### ISSUE TITLE
ADD/COPY --chown cannot determine username when --squash is used

### ISSUE THREAD

**Issue by oblitum** [containers/buildah#1806]
**Description**

Using `--squash` cause `COPY --chown` not to know about a user I've created in a previous step, inside the image.

Related issue: #416.

**Steps to reproduce the issue:**
1. Full `Dockerfile` used (`archlinux/base:latest` has been built and installed locally from sources because it's outdated on DockerHub until next month):
   ```
   FROM archlinux/base:latest
   
   RUN pacman -Sy && pacman -S --noconfirm \
       aria2 aspell aspell-en aspell-pt atool autofs base-devel bat bind-tools \
       cabextract clamav clang cryptsetup devtools dhcpcd dictd dosfstools \
       dvd+rw-tools easy-rsa ethtool exa exfat-utils extra-cmake-modules fd \
       ffmpeg fzf gdb geoip geoip-database-extra gifsicle git go gtop htop hub \
       hugo iftop imagemagick inetutils iotop iperf3 iproute2 iputils iw \
       jfsutils jdk-openjdk jq languagetool lastpass-cli libqalculate \
       logrotate lsof luarocks man-db man-pages mediainfo moreutils msmtp \
       mtools ncdu neomutt netctl newsboat nfs-utils ninja nmap nodejs ntfs-3g \
       ntp openmp openmpi openssh p7zip parallel parted pax-utils peg pigz \
       postgresql powertop proxychains-ng psmisc pygmentize python \
       python-neovim python-pip python-setuptools python-virtualenv python2 \
       python2-pip python2-setuptools python2-virtualenv r radare2 rclone \
       reflector reiserfsprogs ripgrep rtv ruby rubygems rustup sdcv setconf \
       smartmontools strace syncthing termtosvg testdisk texlive-langchinese \
       texlive-langcyrillic texlive-langextra texlive-langgreek \
       texlive-langjapanese texlive-langkorean texlive-most tig tldr tmux tor \
       traceroute translate-shell transmission-cli trash-cli typescript ufw \
       unrar usbutils valgrind vi vint w3m watchexec weechat wget whois woff2 \
       xclip xdg-user-dirs xdg-utils xfsprogs yarn youtube-dl zsh \
    && pacman -Sc --noconfirm \
    && rm -rf /tmp/* \
    && ln -sf /usr/share/zoneinfo/America/Sao_Paulo /etc/localtime
   
   RUN groupadd -g 1000 francisco \
    && useradd -g francisco -G wheel -u 1000 -ms /bin/zsh francisco \
    && sed -i -e 's/^# %wheel ALL=(ALL) NOPASSWD: ALL$/%wheel ALL=(ALL) NOPASSWD: ALL/' /etc/sudoers
   
   RUN su francisco -c "git clone --depth 1 https://aur.archlinux.org/yay.git /tmp/yay && cd /tmp/yay && makepkg -cs" \
    && cd /tmp/yay && pacman -U --noconfirm *.pkg.tar.xz && cd .. && rm -rf yay
   
   USER francisco
   WORKDIR /home/francisco
   
   COPY --chown=francisco:francisco dotfiles .
   
   RUN xdg-user-dirs-update \
    && git config status.showUntrackedFiles no \
    && git clone https://github.com/tmux-plugins/tpm .tmux/plugins/tpm \
    && mkdir .go \
    && mkdir .cargo \
    && curl -fsSL https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
    && (curl -fsSL https://github.com/rbenv/rbenv-installer/raw/master/bin/rbenv-installer | bash || true) \
    && rustup toolchain install stable \
    && rustup default stable \
    && rm -rf /tmp/* .cache
   
   RUN yay -S --removemake --noconfirm \
       antibody-bin bitlbee-libpurple ccls dict-gcide dict-wikt-en-all \
       goobook-git grv mdcat megatools modd neo4j-community neovim-nightly \
       ovpngen pandoc-bin patat-bin pgformatter-git ptpython purescript-bin \
       purple-hangouts-hg python-grip ranger-git sfnt2woff speedometer \
       stardict-oald stardict-wordnet uni2ascii universal-ctags-git \
    && yay -Sc --noconfirm \
    && rm -rf /tmp/* .cache .cargo/*
   
   RUN rustup component add rust-src rls rustfmt \
    && yarn global add mathjs \
    && yarn global add commitizen \
    && yarn global add bash-language-server \
    && yarn global add dockerfile-language-server-nodejs \
    && yarn global add esy \
    && yarn global add bs-platform \
    && yarn global add purescript-language-server \
    && yarn global add spago \
    && pip install --user tmuxp==1.3.2 \
    && pip install --user pgcli \
    && gem install sqlint \
    && rm -rf /tmp/* .cache
   
   COPY --chown=bitlbee:bitlbee bitlbee /var/lib/bitlbee
   
   CMD ["tmux", "new", "-s", "TMUX"]
   ```
2. `podman build --squash -t archlinux/oblitum .`

**Describe the results you received:**

```
STEP 7: COPY --chown=francisco:francisco dotfiles .
Error: error building at STEP "COPY --chown=francisco:francisco dotfiles .": error determining run uid: user: unknown user error looking up user "francisco"
```

**Describe the results you expected:**

Build without issues, like when building without `--squash`.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
[I] ➜ pacman -Q buildah
buildah 1.10.1-1
```

**Output of `buildah version`:**

```
[I] ➜ buildah version
Version:         1.10.1
Go Version:      go1.12.8
Image Spec:      1.0.1
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  v0.7.1
Git Commit:      8c1c2c58
Built:           Sat Aug 17 16:41:23 2019
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
[I] ➜ podman version
Version:            1.5.1
RemoteAPI Version:  1
Go Version:         go1.12.8
OS/Arch:            linux/amd64
```

**Output of `cat /etc/*release`:**

```
[I] ➜ cat /etc/*release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```

**Output of `uname -a`:**

```
[I] ➜ uname -a
Linux leibniz 5.2.9-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 16 11:29:43 UTC 2019 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
[N] ➜ cat ~/.config/containers/storage.conf
[storage]
  driver = "overlay"
  runroot = "/run/user/1000"
  graphroot = "/home/francisco/.local/share/containers/storage"
  [storage.options]
    size = ""
    remap-uids = ""
    remap-gids = ""
    ignore_chown_errors = ""
    remap-user = ""
    remap-group = ""
    ostree_repo = ""
    skip_mount_home = ""
    mount_program = "/usr/bin/fuse-overlayfs"
    mountopt = ""
    [storage.options.thinpool]
      autoextend_percent = ""
      autoextend_threshold = ""
      basesize = ""
      blocksize = ""
      directlvm_device = ""
      directlvm_device_force = ""
      fs = ""
      log_level = ""
      min_free_space = ""
      mkfsarg = ""
      mountopt = ""
      use_deferred_deletion = ""
      use_deferred_removal = ""
      xfs_nospace_max_retries = ""
```

**Comment by bebosudo** [comment]
I've seen the same issue. Here's a shorter repro based on a jupyter base image which defines a var `NB_USER`:
```dockerfile
FROM jupyter/base-notebook:hub-1.1.0

RUN env
COPY --chown=${NB_USER}:users stuff /stuff
```

When building, `NB_USER` is visible inside the env, but podman breaks on the copy with --chown:
```
STEP 1: FROM jupyter/base-notebook:hub-1.1.0
STEP 2: RUN env
NB_USER=jovyan
NB_UID=1000
...
STEP 3: COPY --chown=${NB_USER}:users stuff /stuff
Error: error building at STEP "COPY --chown=${NB_USER}:users stuff /stuff": error determining run uid: user: unknown user error looking up user "${NB_USER}"
```

**Comment by bebosudo** [comment]
Podman 1.8.1 (just built for centos 7 [here](https://github.com/bebosudo/containers-rpm/commit/c0718e4c606c2c18822d9097d63021036b8e4365)) is also affected; docker instead builds smoothly.

**Comment by rhatdan** [comment]
Opened a PR to fix this here
https://github.com/openshift/imagebuilder/pull/149

**Comment by oblitum** [comment]
Hi, sorry because at the moment I can't test the PR.

**Comment by bebosudo** [comment]
I confirm that PR https://github.com/openshift/imagebuilder/pull/149 is working fine.

I patched podman 1.8.1 with the following patch, which skips sections missing from 149:
```patch
diff -urN libpod-1.8.1_orig/vendor/github.com/openshift/imagebuilder/dispatchers.go libpod-1.8.1/vendor/github.com/openshift/imagebuilder/dispatchers.go
--- libpod-1.8.1_orig/vendor/github.com/openshift/imagebuilder/dispatchers.go	2020-03-11 15:16:13.000000000 +0100
+++ libpod-1.8.1/vendor/github.com/openshift/imagebuilder/dispatchers.go	2020-03-15 00:31:04.307075646 +0100
@@ -132,7 +132,11 @@
 	last := len(args) - 1
 	dest := makeAbsolute(args[last], b.RunConfig.WorkingDir)
 	if len(flagArgs) > 0 {
-		for _, arg := range flagArgs {
+        for _, a := range flagArgs {
+            arg, err := ProcessWord(a, b.Env)
+            if err != nil {
+                return err
+            }
 			switch {
 			case strings.HasPrefix(arg, "--chown="):
 				chown = strings.TrimPrefix(arg, "--chown=")
@@ -158,7 +162,12 @@
 	var chown string
 	var from string
 	if len(flagArgs) > 0 {
-		for _, arg := range flagArgs {
+        for _, a := range flagArgs {
+            arg, err := ProcessWord(a, b.Env)
+            if err != nil {
+                return err
+            }
+
 			switch {
 			case strings.HasPrefix(arg, "--chown="):
 				chown = strings.TrimPrefix(arg, "--chown=")
```

And this is the repro now running fine as with docker:
```bash
$ > /tmp/stuff
$ podman build -f /tmp/Dockerfile /tmp
STEP 1: FROM jupyter/base-notebook:hub-1.1.0
STEP 2: RUN env
--> Using cache f2ca8baa9e56bab9d37cb694d23b9d8dc819aef6d75a5c0557cd8fd6065157a7
STEP 3: COPY --chown=${NB_USER}:users stuff /stuff
STEP 4: COMMIT
--> 5ac1a622259
5ac1a622259cf8bd2e7e2da70ed57b752c49a4b3afa48a4544ab9114f2a6a00c
$
```

Thanks!

**Comment by rhatdan** [comment]
I believe this is fixedin master branch, Reopen if I am mistaken.
